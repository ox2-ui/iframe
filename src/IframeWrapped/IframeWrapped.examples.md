### IframeWrapped

    const styles = {
      wrapper: {
        width: '200px',
        height: '300px',
      },
    };

    <div style={styles.wrapper}>
      <IframeWrapped url="http://example.com" />
    </div>
