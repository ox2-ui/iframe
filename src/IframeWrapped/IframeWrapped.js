import PropTypes from 'prop-types';
import React from 'react';
import Iframe from '../Iframe';

const styles = {
  wrapper: {
    position: 'absolute',
    top: '0',
    left: '0',
    right: '0',
    bottom: '0',
  },
};

/**
 * IframeWrapped Component
 */
const IframeWrapped = (props) => {
  const { className, style, url } = props;

  const mergedStyles = {
    ...styles.wrapper,
    ...style,
  };

  return (
    <div
      className={className}
      style={mergedStyles}
    >
      <Iframe url={url} />
    </div>
  );
};

IframeWrapped.propTypes = {
  /**
   * The css class name of the root element.
   */
  className: PropTypes.string,
  /**
   * The css style object of the root element.
   */
  style: PropTypes.object,
  /**
   * Url to load into the iframe
   */
  url: PropTypes.string,
};

export default IframeWrapped;

