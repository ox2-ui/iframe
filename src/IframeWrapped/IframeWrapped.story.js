import React from 'react';
import { storiesOf } from '@storybook/react';
import IframeWrapped from './IframeWrapped';

storiesOf('IframeWrapped', module).add('default', () => (
  <IframeWrapped url={'sample/index.html'} />
));
