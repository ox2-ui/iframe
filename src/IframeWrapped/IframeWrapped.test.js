/* eslint-disable no-unused-vars */
import React from 'react';
import { shallow, render, mount } from 'enzyme';
import IframeWrapped from './IframeWrapped';

it('renders correctly', () => {
  const wrapper = mount(
    <IframeWrapped url={'sample/index.html'} />
  );

  expect(wrapper).toMatchSnapshot();
});
