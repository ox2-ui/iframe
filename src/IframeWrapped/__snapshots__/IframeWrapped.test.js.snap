// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<IframeWrapped
  className=""
  style={Object {}}
  url="sample/index.html"
>
  <div
    className=""
    style={
      Object {
        "bottom": "0",
        "left": "0",
        "position": "absolute",
        "right": "0",
        "top": "0",
      }
    }
  >
    <Iframe
      className=""
      style={Object {}}
      url="sample/index.html"
    >
      <AutoSizer
        onResize={[Function]}
      >
        <div
          style={
            Object {
              "height": 0,
              "overflow": "visible",
              "width": 0,
            }
          }
        >
          <iframe
            className=""
            height={0}
            src="sample/index.html"
            style={
              Object {
                "border": "none",
                "margin": "0",
                "padding": "0",
              }
            }
            width={0}
          >
            <p>
              Your browser does not support iframes.
            </p>
          </iframe>
        </div>
      </AutoSizer>
    </Iframe>
  </div>
</IframeWrapped>
`;
